### PR TITLE
LibJS: Typed array constructor fixes

### PR DESCRIPTION
**Pull request by IdanHo**
This fixes 13 test262 test cases.

### CODE REVIEW

**Comment by linusg** [comment]
Let's capitalize the first letter of the AOs to match the spec, i.e. GetPrototypeFromConstructor and GetMethod. Also, build error :)

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp [hunk 1]
```diff
@@ -257,15 +266,15 @@ void TypedArrayBase::visit_edges(Visitor& visitor)
...
         if (vm.argument_count() == 0)                                                                                                  \
-            return ClassName::create(global_object(), 0);                                                                              \
+            return ClassName::create(global_object(), 0, new_target);                                                                  \
                                                                                                                                        \
         auto first_argument = vm.argument(0);                                                                                          \
         if (first_argument.is_object()) {                                                                                              \
-            auto* typed_array = ClassName::create(global_object(), 0);                                                                 \
+            auto* typed_array = ClassName::create(global_object(), 0, new_target);                                                     \
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArray.cpp:283]: This probably needs an exception check now

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp [hunk 2]
```diff
@@ -143,6 +143,12 @@ static void initialize_typed_array_from_array_like(GlobalObject& global_object,
     if (vm.exception())
         return;
 
+    // Enforce 2GB "Excessive Length" limit
+    if (length > NumericLimits<i32>::max() / sizeof(TypeError)) {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/TypedArray.cpp:147]: not sure I understand this, why `sizeof(TypeError)`?

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/TypedArray.cpp:283]: yep, fixed

File: Userland/Libraries/LibJS/Runtime/TypedArray.cpp (hunk 2 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/TypedArray.cpp:147]: Oops! i meant `Type`, fixed!

**Pull request merged**